### ISSUE TITLE
Force uploaded image to be square

### ISSUE THREAD

**Issue by tedsecretsource** [Laravel-Backpack/CRUD#1150]
Not a bug report, but a question…

How can I force images uploaded via the image type input to be square?

Given a user uploads an image
- And the input is specified as type "image"
- And the height and width of the image are not the same

When the image is being processed by the JavaScript library that handles the preview (ideally) or in PHP after the form has been submitted (acceptable)

Then an error is returned
- And the user is prompted to upload a square image

What is the recommended procedure for doing this? Also, where can I find more documentation on the JS library you're using for the image preview and cropping (looks great, BTW).

Thanks in advance.

Ted Stresen-Reuter

**Comment by tabacitu** [comment]
Hi @tedsecretsource ,

You have solutions for both frond and back:
- specify aspect ratio 1 for the [image field](https://laravel-backpack.readme.io/docs/crud-fields#section-image);
- if you use another field type (not image), you can use [Intervention Image](http://image.intervention.io/) to make the image square in your upload accessor;

Hope it helps.
Cheers!

**Comment by tedsecretsource** [comment]
Hi.

Thanks for the fast response. I just tried setting the aspect_ratio to 1, expecting it to force the cropping to maintain the aspect ratio of the original image (which was a rectangle in this case) and to my delight, it was a square!

I then tried a few different values and realized that the number is not a boolean but simply a value that represents a ratio. Very slick, very helpful, very cool!

Thanks a lot. If I were to update the documentation, I would add the following:

> **A note about aspect_ratio**
> The value for aspect ratio is a float that represents the ratio of the cropping rectangle height and width. By way of example, 
> * Square = 1
> * Landscape = 2
> * Portrait = 0.5
> 
> And you can, of course, use any value for more extreme rectangles.

HTH and thanks for the great product.

Kind regards,

Ted Stresen-Reuter
